### PR TITLE
CLC-5372 - Web Accessibility 3: misc accessibility issues

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.ets.berkeley.edu/jira/browse/CLC-5372

Adds focus management, accessibility announcement service, and other misc accessibility tweaks.

### CODE REVIEW

**Comment by raydavis** [comment]
@redconfetti , thank you for the hard work. Although this PR is a bit more sprawling than usual, given the Monday review I see why it makes sense -- we'll be more granular in the follow-ups.

File: src/assets/javascripts/angular/services/utilService.js [hunk 1]
```diff
@@ -169,8 +169,31 @@
 
     var uidPattern = /^[0-9]{1,9}$/;
 
+    var accessibilityAnnounce = function(message) {
+      // remove existing announcer
+      var existingAnnouncer = $window.document.getElementById('cc-sr-announcer');
```
Comment (christianv) [inline src/assets/javascripts/angular/services/utilService.js]: @redconfetti can we move this to a directive? We're accessing the DOM

File: src/assets/javascripts/angular/services/utilService.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 (function(angular, naturalSort) {
   'use strict';
 
-  angular.module('calcentral.services').service('utilService', function($cacheFactory, $http, $location, $rootScope, $route, $window) {
+  angular.module('calcentral.services').service('utilService', function($cacheFactory, $http, $location, $rootScope, $route, $timeout, $window) {
```
Comment (christianv) [inline src/assets/javascripts/angular/services/utilService.js]: @redconfetti is `$timeout` used anywhere in this file?

Comment (redconfetti) [inline src/assets/javascripts/angular/services/utilService.js]: @christianv No. I just removed it in #3926

File: src/assets/javascripts/angular/services/utilService.js (hunk 1 above)
Comment (redconfetti) [inline src/assets/javascripts/angular/services/utilService.js]: I can see you created Jira [CLC-5436](https://jira.ets.berkeley.edu/jira/browse/CLC-5436) concerning this. I'll try to take care of this during this sprint.

Comment (christianv) [inline src/assets/javascripts/angular/services/utilService.js]: @redconfetti thanks

**Pull request merged**